src/main/java/com/marklogic/appdeployer/command/security/InsertCertificateHostsTemplateCommand.java

```diff
@@ -76,12 +76,15 @@ protected File determinePrivateKeyFile(File publicCertificateFile) {
 	/**
 	 * @param context
 	 * @param templateName The name of the certificate template that the host certificate will be inserted into
+	 * 	Assumes filename is hostname + .crt: ex: host1.marklogic.com.crt
```

Comment:
Assuming this is the case by default would be breaking behavior for users whose certificate filenames do not conform to this pattern. 

The problem here seems to be - when checking to see if a certificate exists, we may need to account for not just the name of the template but the host-name of the template as well. 

Since that's a "may", then that means a user needs a way to enable/disable this feature. And to avoid breaking changes, we would need it disabled by default. 

The next challenge is - given a File, how do we determine the host-name for that certificate? A naming convention is one mechanism - e.g. "(host-name).crt". It may suffice for this feature to say - look, if you want ml-app-deployer to account for the host-name when checking to see if a certificate exists, then your filename must match that pattern. 

So I think we need addition here for sure - a boolean on this class that controls whether host-name is used. And that boolean has to default to false to preserve backwards compatibility. 

If the boolean is true, then I think it's fine in the short-term to require the above naming convention. 

I'll submit a little PR into this one, including some test coverage as well, based on the above thinking.

Reply:
So IIRC, ML requires that a certificate being installed match a hostname (ie "CN of cert" = hostname in cluster).  So I think it is a safe assumption/requirement that the user name the crt and key files to match.  This was my original intention when I first contributed the insertion code.

So, to solve the now "breaking change" issue - (although I would rather not allow users to name their crt files whatever they want - leaving ml-app-deployer with spaghetti code trying to sort it out) - we **could** use: xdmp:x509-certificate-extract (or similar) call to determine the CN of each cert in advance of insertion.  I don't think a boolean property is the answer here.

Reply:
I like the idea of using x509-certificate-extract - to confirm, the value we're interested in from the output of that function is /cert/subject/commonName (ignoring namespace prefixes)?

Would it seem reasonable then for this command to say - okay, I'm going to make a call to ML and use the x509 function to extract the commonName. Assuming I get a value back, I will then use it when checking to see if the cert already exists. If anything goes wrong, I'll log a warning and state that I'll be checking to see if the cert exists but without looking for the host name.

How does that sound?

Reply:
yes sounds good. - how can I assist?

Reply:
I'll update the PR I created, it'll be simple to do - thanks for getting this started and the good idea as well.